### PR TITLE
3.4 driver compat diagram updates

### PR DESCRIPTION
**Pull request by schmalliso**
- 3.4 driver compat: C, Cxx, Node, Java, Csharp
- 3.4 driver compat: perl
- 3.4 driver compat: php
- 3.4 driver compat: python
- 3.4 driver compat: motor
- 3.4 driver compat: ruby
- 3.4 driver compat: scala
- 3.4: driver compat cleanup

Staged: https://docs-mongodbcom-staging.corp.mongodb.com/allison/master/drivers/driver-compatibility-reference.html

### CODE REVIEW

File: source/drivers/driver-compatibility-reference.txt [hunk 1]
```diff
@@ -1077,27 +1183,69 @@ MongoDB Compatibility
      - MongoDB 2.4
      - MongoDB 2.6
      - MongoDB 3.0
+     - MongoDB 3.2
+     - MongoDb 3.4
```
Comment (behackett) [inline source/drivers/driver-compatibility-reference.txt]: @ajdavis can you comment on this? Should we clarify that Motor supports MongoDB 3.4 when using PyMongo 3.4? And is using PyMongo 3.4 enough to fully support all new features in MongoDB 3.4?

Comment (behackett) [inline source/drivers/driver-compatibility-reference.txt]: MongoDb -> MongoDB

File: source/drivers/driver-compatibility-reference.txt [hunk 2]
```diff
@@ -1077,27 +1183,69 @@ MongoDB Compatibility
...
 
    * - 0.1
      - |checkmark|
      -
      -
+     -
+     -
 
+- Motor 1.0 wraps PyMongo 3.3+
+- Motor 0.7 wraps PyMongo 2.9.4+
```
Comment (behackett) [inline source/drivers/driver-compatibility-reference.txt]: Motor 0.7 wraps PyMongo >= 2.9.4  and < 3.0. That is, it does not work with PyMongo 3.x.

File: source/drivers/driver-compatibility-reference.txt [hunk 3]
```diff
@@ -1121,56 +1269,100 @@ Language Compatibility
...
      - |checkmark|
      - |checkmark|
+     - |checkmark|
      -
      -
-     - |checkmark|
      -
 
+- Motor 1.0 wraps PyMongo 3.3+
+- Motor 0.7 wraps PyMongo 2.9.4+
```
Comment (behackett) [inline source/drivers/driver-compatibility-reference.txt]: Same comment here about not requiring PyMongo < 3.0.

**Comment by christkv** [comment]
@schmalliso I shared the matrix spreadsheet with you that contains the info for the node driver.

File: source/drivers/driver-compatibility-reference.txt [hunk 4]
```diff
@@ -645,56 +699,70 @@ MongoDB Compatibility
...
    :class: compatibility
 
    * - PHP Driver
      - MongoDB 2.4
      - MongoDB 2.6
      - MongoDB 3.0
      - MongoDB 3.2
+     - MongoDB 3.4
+
+   * - mongodb-1.2
```
Comment (jmikola) [inline source/drivers/driver-compatibility-reference.txt]: This should say: "PHPLIB 1.1 + mongodb-1.2"

File: source/drivers/driver-compatibility-reference.txt [hunk 5]
```diff
@@ -718,6 +786,14 @@ Language Compatibility
      - PHP 7.0
      - HHVM 3.9
 
+   * - mongodb-1.2
+     -
+     - |checkmark|
+     - |checkmark|
+     - |checkmark|
+     - |checkmark|
+     - |checkmark|
```
Comment (jmikola) [inline source/drivers/driver-compatibility-reference.txt]: PHP 7.1's release date is December 1 (see: [here](https://wiki.php.net/todo/php71)), so you might as well add a column for that.

@derickr can chime in if additional HHVM versions should be listed. I believe 3.15 is the most recent LTS (3.9 being the last), and that is supported by the current HHVM driver.

Comment (derickr) [inline source/drivers/driver-compatibility-reference.txt]: Yes, 3.15 is the latest LTS. HHVM 3.12 was also an LTS release, and is also supported. 3.13 and 3.14 don't have to be mentioned, although they do work. 

The HHVM driver works with HHVM 3.12 and 3.15, and I'm not sure where 3.9 came from, as I think we never supported that.

File: source/drivers/driver-compatibility-reference.txt (hunk 1 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: Thanks! Fixed.

File: source/drivers/driver-compatibility-reference.txt (hunk 2 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: Done!

File: source/drivers/driver-compatibility-reference.txt (hunk 3 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: Done!

File: source/drivers/driver-compatibility-reference.txt [hunk 6]
```diff
@@ -1121,56 +1269,100 @@ Language Compatibility
      - Python 2.5
      - Python 2.6
      - Python 2.7
-     - Python 3.1
-     - Python 3.2
      - Python 3.3
      - Python 3.4
+     - Python 3.5
+     - Python 3.6
```
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: Cool. I'll remove it - I stole the language compatibility for Motor from the http://motor.readthedocs.io/en/stable/requirements.html so y'all may want to update that as well 😺.

File: source/drivers/driver-compatibility-reference.txt (hunk 4 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: Sweet. I suspected as much but was having trouble digging through to figure out the canonical answer. Thanks!

File: source/drivers/driver-compatibility-reference.txt (hunk 5 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: @jmikola/@derickr Which driver versions support PHP 7.1 (so I can put the appropriate checkmarks)? I assume the same as support PHP7, but best to be sure :)

**Comment by schmalliso** [comment]
@christkv @behackett @jmikola Updated! Thanks folks 😺

**Comment by estolfo** [comment]
Hi, ruby needs some updates as well:

The versions 2.3 and 2.4 need to be added to the Ruby Driver column. They both support server versions 2.4 - 3.4
MongoDB 3.4 needs to be added to the top row.

Our language compatibility matrix needs to add to the Ruby driver column:

2.1, 2.2, 2.3, 2.4
and change "1.12 - 1.9" to say "1.9 - 1.12"

To the top row, we need to add:
Ruby 2.2 and Ruby 2.3

**Comment by estolfo** [comment]
ruby part LGTM

File: source/drivers/driver-compatibility-reference.txt (hunk 5 above)
Comment (schmalliso) [inline source/drivers/driver-compatibility-reference.txt]: LGTM'd by Jeremy in Slack.

**Comment by schmalliso** [comment]
We've merged this in so we can publish this morning with the release. If you have further revisions, please file a new ticket for your driver! Thanks

**Pull request merged**